### PR TITLE
Hxb writer config

### LINKED ISSUE #11490: [hxb] Filter for --hxb output
One thing I've been ignoring so far is that we currently can't really specify what actually goes into a generated .hxb file. For library usage, we typically don't want to include code from std and such, so it would be nice to be able to configure this somehow.

It's probably not super urgent because I'm the kind of guy who will just tell someone to delete things from the generated zip file... but maybe usability could be a little better than that.

### PR DESCRIPTION
**Pull request by Simn**
Allows specifying `--hxb some.json` and adds `haxe.macro.Compiler.getHxbConfiguration` and `haxe.macro.Compiler.setHxbConfiguration`.

All the pertinent information should be in the .hx files. @PXshadow Could you check if this works for your use-case?

Closes #11490

### CODE REVIEW

File: src/compiler/args.ml [hunk 1]
```diff
@@ -278,9 +278,9 @@ let parse_args com =
 		("Services",["--json"],[],Arg.String (fun file ->
 			actx.json_out <- Some file
 		),"<file>","generate JSON types description");
-		("Services",["--hxb"],[], Arg.String (fun dir ->
-			actx.hxb_out <- Some dir;
-		),"<directory>", "generate haxe binary representation in target directory");
```
Comment (kLabz) [inline src/compiler/args.ml:283]: Might be nice to rename the parameter too? :grin:

**Comment by Simn** [comment]
Where would be put one of these fancy JSON schema things for this?

(And is there a tool to generate those from a type definition?)

**Comment by kLabz** [comment]
json2object lib can generate them: https://github.com/elnabo/json2object/#using-the-jsonschema-writer
I don't remember where we put it, checking..

Edit:
Haxelib.json schema is there: https://github.com/HaxeFoundation/haxelib/blob/master/schema.json

Guess we could start storing some here in `extra/schema` or something?
We will also need to publish those [on schemastore](https://github.com/SchemaStore/schemastore) when ready ([example](https://github.com/SchemaStore/schemastore/pull/2621/files))

**Comment by kLabz** [comment]
Currently we have schemas there, which isn't ideal imo but better than nothing

https://github.com/vshaxe/vshaxe/tree/master/schemas

**Comment by Simn** [comment]
We have `.vscode/schemas` but I vaguely remember that this wasn't the right place either, and it's also gitignored.

Edit: On the bright side, looks like json2object just works:

```
{
	"definitions": {
		"haxe.hxb.WriterTargetConfig": {
			"additionalProperties": false,
			"properties": {
				"include": {
					"description": "Dot paths of modules or packages to be included in the archive. This takes priority over exclude. By default, all modules that aren't explicitly excluded are included.",
					"items": {
						"type": "string"
					},
					"type": "array"
				},
				"hxbVersion": {
					"description": "The hxb version to target. By default, the version of the Haxe compiler itself is targeted. See https://github.com/HaxeFoundation/haxe/issues/11505",
					"type": "integer"
				},
				"generateDocumentation": {
					"description": "If false, no documentation",
					"type": "boolean"
				},
				"generate": {
					"description": "If `false`, this target is ignored by the writer.",
					"type": "boolean"
				},
				"exclude": {
					"description": "Dot paths of modules or packages to be exluded from the archive.",
					"items": {
						"type": "string"
					},
					"type": "array"
				}
			},
			"type": "object"
		},
		"haxe.hxb.WriterConfig": {
			"additionalProperties": false,
			"properties": {
				"targetConfig": {
					"description": "The configuration for the current target context. If it is `null`, all data for the target context is generated.",
					"$ref": "#/definitions/haxe.hxb.WriterTargetConfig"
				},
				"macroConfig": {
					"description": "The configuration for the macro context. If it is `null`, all data for the macro context is generated.",
					"$ref": "#/definitions/haxe.hxb.WriterTargetConfig"
				},
				"archivePath": {
					"description": "The file path for the archive. Occurrences of `$target` are replaced by the name of the current target (js, hl, etc.).",
					"type": "string"
				}
			},
			"required": [
				"archivePath"
			],
			"type": "object"
		}
	},
	"$ref": "#/definitions/haxe.hxb.WriterConfig",
	"$schema": "http://json-schema.org/draft-07/schema#"
}
```

**Comment by kLabz** [comment]
Those two might work for Haxe repo itself (... on vscode only)
They're likely outdated, and one should be added for warnings..

We could move all those schemas to `extra` with the generator code there too? So we don't update those manually (and forget to do it)

**Comment by Simn** [comment]
Ok I see, I shouldn't have brought that up here... made a new issue!

**Comment by Simn** [comment]
I'll go ahead and merge this. The foundation should be good, and if we need anything else it's easy enough to add thanks to the new internal data API here. The macro side of this isn't very well tested (read: not tested at all) but I'm sure people will get to that.

**Pull request merged**